openhands/sdk/llm/message.py

```diff
@@ -195,18 +254,58 @@ def from_litellm_message(cls, message: LiteLLMMessage) -> "Message":
 
         Provider-agnostic mapping for reasoning:
         - Prefer `message.reasoning_content` if present (LiteLLM normalized field)
+        - Extract `thinking_blocks` from content array (Anthropic-specific)
         """
         assert message.role != "function", "Function role is not supported"
 
         rc = getattr(message, "reasoning_content", None)
+        content = []
+        thinking_blocks = []
+
+        # Handle content
+        if isinstance(message.content, str):
+            content = [TextContent(text=message.content)]
+        elif isinstance(message.content, list):
+            # Process content list and extract thinking blocks
+            for item in message.content:  # type: ignore[union-attr]
+                if isinstance(item, dict):
+                    item_type = item.get("type", "")
+                    if item_type == "thinking":
+                        # Extract thinking block
+                        thinking_content = item.get("thinking", "")
+                        signature = item.get("signature")
+                        thinking_block = ThinkingBlock(
+                            thinking=thinking_content,
+                            signature=signature,
+                        )
+                        thinking_blocks.append(thinking_block)
+                    elif item_type == "text":
+                        # Extract text content
+                        text_content = TextContent(text=item.get("text", ""))
+                        content.append(text_content)
+
+        # Extract thinking blocks from LiteLLM's thinking_blocks field (fallback)
+        litellm_thinking_blocks = getattr(message, "thinking_blocks", None)
+        if litellm_thinking_blocks:
+            for block in litellm_thinking_blocks:
+                # Convert LiteLLM thinking block to our ThinkingBlock
+                # LiteLLM thinking blocks are TypedDict objects
+                thinking_content = (
+                    block.get("thinking", "") if isinstance(block, dict) else ""
+                )
+                signature = block.get("signature") if isinstance(block, dict) else None
+                thinking_block = ThinkingBlock(
+                    thinking=thinking_content,
+                    signature=signature,
+                )
+                thinking_blocks.append(thinking_block)
```

Comment:
This also feels wrong - litellm supposed to clean this up for us, we should just set content to content and thinking_blocks to thinking blocks